src/gc.c

```diff
@@ -163,6 +163,11 @@ void GCContext_Stop(GCContext* gc) {
   if (RedisModule_StopTimer(ctx, gc->timerID, (void**)&data) == REDISMODULE_OK) {
     assert(data->gc == gc);
     rm_free(data);  // release task memory
+
+    // free gc
+    gc->callbacks.onTerm(gc->gcCtx);
+    rm_free(gc);
+    return;
```

Comment:
This is what actually makes the difference.
If a task is found in Redis Timer, we know the GC is not currently running so we can free the GC context right away instead of passing it to the thread to be released.